pkg/authn/identityheaders/identityheaders.go

```diff
@@ -52,7 +53,31 @@ func WithAuthHeaders(handler http.Handler, cfg *AuthnHeaderConfig) http.Handler
 			// Seemingly well-known headers to tell the upstream about user's identity
 			// so that the upstream can achieve the original goal of delegating RBAC authn/authz to kube-rbac-proxy
 			req.Header.Set(cfg.UserFieldName, u.GetName())
-			req.Header.Set(cfg.GroupsFieldName, strings.Join(u.GetGroups(), cfg.GroupSeparator))
+
+			if cfg.GroupSeparator == "" {
+				// Add each group as a separate header entry
+				for _, group := range u.GetGroups() {
```

Comment:
turn the groups dropping logic into a function and reuse it in both branches

Reply:
updated, thank you :)